src/ui.rs

```diff
@@ -140,11 +140,11 @@ impl<'stdout> Ui<'stdout> {
     }
 
     pub fn write(&mut self, text: &str) -> io::Result<()> {
-        self.stdout_formatter().write_str(text)
+        self.stdout.get_mut().unwrap().write_all(text.as_bytes())
```

Comment:
This works with the two formatters (plain and color), but would not work e.g. if the color-formatter emitted the color code for each string and cleared it afterwards, for example (right?). I think that's a fair assumption to make, but I wanted to point it out in case we ever break that assumption.

Reply:
Resolving this since I'm not asking for any changes.

Reply:
`ui.stdout_formatter()` immutably borrows `ui`, so `write(&mut self)` is blocked while the formatter is alive.

If the formatter is sexp-formatter, and if `ui.write()` is supposed to be parenthesized by default as well as `formatter().write_str()`, right, my assumption is no longer valid.